### PR TITLE
app-editors/texworks: Fix call to undeclared function vasprintf

### PR DESCRIPTION
**Pull request by listout**
Closes: https://bugs.gentoo.org/882461

### CODE REVIEW

**Comment by thesamesam** [comment]
This is why it doesn't show up with GCC: https://github.com/TeXworks/texworks/commit/ccf692bb38c43a38f8e9f755dd48451c6f6ad430.

That is wrong - let's just drop the wrong if() there.

**Comment by listout** [comment]
@thesamesam is the current patch correct?

File: app-editors/texworks/files/texworks-0.6.7-vasprintf-clang16-fix.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
++++ b/modules/synctex/CMakeLists.txt
+@@ -23,9 +23,3 @@ target_link_libraries(SyncTeX ZLIB::ZLIB)
+ if (WIN32)
+ 	target_link_libraries(SyncTeX shlwapi)
+ endif()
+-
+-# SyncTeX uses vasprintf which is not in POSIX; thus gcc prints a warning unless
+-# _GNU_SOURCE is defined. Other compilers seem to be fine.
+-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+-  target_compile_definitions(SyncTeX PRIVATE _GNU_SOURCE)
```
Comment (thesamesam) [inline app-editors/texworks/files/texworks-0.6.7-vasprintf-clang16-fix.patch:13]: Keep the target_ line, just drop the if

Comment (listout) [reply app-editors/texworks/files/texworks-0.6.7-vasprintf-clang16-fix.patch:10]: Done, thank you.

**Review by juippis — COMMENTED** [review]
Uuh...
```
 * Package:    app-editors/texworks-0.6.7-r1:0
 * Repository: gentoo
 * USE:        abi_x86_64 amd64 elibc_musl kernel_linux lua_single_target_lua5-1 python_single_target_python3_11 test
 * FEATURES:   network-sandbox preserve-libs sandbox test userpriv usersandbox
 * Using python3.11 to build
>>> Unpacking source...
>>> Unpacking texworks-0.6.7.tar.gz to /var/tmp/portage/app-editors/texworks-0.6.7-r1/work
>>> Source unpacked in /var/tmp/portage/app-editors/texworks-0.6.7-r1/work
>>> Preparing source in /var/tmp/portage/app-editors/texworks-0.6.7-r1/work/texworks-release-0.6.7 ...
 * Source directory (CMAKE_USE_DIR): "/var/tmp/portage/app-editors/texworks-0.6.7-r1/work/texworks-release-0.6.7"
 * Build directory  (BUILD_DIR):     "/var/tmp/portage/app-editors/texworks-0.6.7-r1/work/texworks-release-0.6.7_build"
 * Applying texworks-0.6.5-cmake_lua_version.patch ...
 [ ok ]
 * Applying texworks-0.6.7-vasprintf-clang16-fix.patch ...
 [ ok ]
 * Hardcoded definition(s) removed in modules/QtPDF/CMakeLists.txt:
 *  SET(CMAKE_COLOR_MAKEFILE ON)
 *    SET(CMAKE_BUILD_TYPE "Debug")
 * Hardcoded definition(s) removed in CMakeLists.txt:
 *  SET(CMAKE_COLOR_MAKEFILE ON)
 *    SET( CMAKE_INSTALL_PREFIX "/Applications" )
 *    SET(CMAKE_BUILD_TYPE "Release")
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/app-editors/texworks-0.6.7-r1/work/texworks-release-0.6.7 ...
 * Source directory (CMAKE_USE_DIR): "/var/tmp/portage/app-editors/texworks-0.6.7-r1/work/texworks-release-0.6.7"
 * Build directory  (BUILD_DIR):     "/var/tmp/portage/app-editors/texworks-0.6.7-r1/work/texworks-release-0.6.7_build"
cmake -C /var/tmp/portage/app-editors/texworks-0.6.7-r1/work/texworks-release-0.6.7_build/gentoo_common_config.cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -Wno-dev -DPREFER_BUNDLED_SYNCTEX=ON -DWITH_LUA=no -DWITH_PYTHON=no -DWITH_TESTS=yes -DTeXworks_PLUGIN_DIR=/usr/lib/texworks -DTeXworks_DOCS_DIR=/share/doc/texworks-0.6.7-r1 -DQTPDF_VIEWER=ON -DBUILD_SHARED_PLUGINS=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_TOOLCHAIN_FILE=/var/tmp/portage/app-editors/texworks-0.6.7-r1/work/texworks-release-0.6.7_build/gentoo_toolchain.cmake /var/tmp/portage/app-editors/texworks-0.6.7-r1/work/texworks-release-0.6.7
loading initial cache file /var/tmp/portage/app-editors/texworks-0.6.7-r1/work/texworks-release-0.6.7_build/gentoo_common_config.cmake
-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/x86_64-gentoo-linux-musl-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/x86_64-gentoo-linux-musl-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
fatal: not a git repository: '.git'
fatal: not a git repository: '.git'
-- Git commit info updated
-- Found ZLIB: /usr/lib/libz.so (found version "1.2.13")  
-- Found PkgConfig: /usr/bin/x86_64-gentoo-linux-musl-pkg-config (found version "1.8.1") 
-- Found Hunspell: /usr/lib/libhunspell-1.7.so  
-- Generating default binary paths:
   '/usr/local/texlive/2028/bin/x86_64-linux', '/usr/local/texlive/2027/bin/x86_64-linux', '/usr/local/texlive/2026/bin/x86_64-linux', '/usr/local/texlive/2025/bin/x86_64-linux', '/usr/local/texlive/2024/bin/x86_64-linux', '/usr/local/texlive/2023/bin/x86_64-linux', '/usr/local/texlive/2022/bin/x86_64-linux', '/usr/local/texlive/2021/bin/x86_64-linux', '/usr/local/texlive/2020/bin/x86_64-linux', '/usr/local/texlive/2019/bin/x86_64-linux', '/usr/local/texlive/2018/bin/x86_64-linux', '${HOME}/bin', '/usr/local/bin', '/usr/bin'
-- Found Poppler private headers: /usr/include/poppler
CMake Error at /usr/share/cmake/Modules/GetPrerequisites.cmake:846 (message):
  /usr/bin/ldd failed: 127

  Error loading shared library libstdc++.so.6: No such file or directory
  (needed by /usr/lib/libpoppler.so)

  Error loading shared library libgcc_s.so.1: No such file or directory
  (needed by /usr/lib/libpoppler.so)

  Error relocating /usr/lib/libpoppler.so: _Znam: symbol not found

  Error relocating /usr/lib/libpoppler.so:
  _ZSt29_Rb_tree_insert_and_rebalancebPSt18_Rb_tree_node_baseS0_RS_: symbol
  not found
```

**Comment by listout** [comment]
@juippis ***@***.***> which profile/environment? On musl-llvm I got
the following http://ix.io/4Dbt

**Comment by juippis** [comment]
Yeah it is on musl. You are building **app-editors/texworks-0.6.7**, not **app-editors/texworks-0.6.7-r1** in your build.log :) but unfortunately that fails for me too. I guess I'm missing some Qt dependency, just can't figure out what it is. I see you're on stable, while I'm on ~. Can you show build.log for -r1 version?

**Comment by listout** [comment]
My bad, http://ix.io/4Dea is the log for texworks-0.6.7-r1.ebuild

**Review by juippis — APPROVED** [review]
Thanks, let's try it out. Couldn't figure what I'm missing even after **ldd**ing and **scanelf**ing some relevant libraries. Unfortunately cmake isn't helping much either.

**Pull request closed without merging**